src/stream/src/executor/hash_agg.rs

```diff
@@ -306,6 +310,10 @@ impl<K: HashKey, S: StateStore> HashAggExecutor<K, S> {
         let mut buffered = stream::iter(futs).buffer_unordered(10).fuse();
         while let Some(result) = buffered.next().await {
             let (key, agg_group) = result?;
+            // We then know this agg group by key appears the first time
+            if agg_group.prev_row_count() == 0 {
+                total_keys_count.fetch_add(1, Ordering::Relaxed);
```

Comment:
What if RisingWave restarted? What if a group key is deleted?

Reply:
The intention is to calculate how many keys exist right now,
do you mean removing the `first time` in the comment?

> What if RisingWave restarted? 
I expect `prev_row_count` reverts to the last checkpoint, which is a correct number.

> What if a group key is deleted?
if it is because of `Op::Delete`, then prev_row_count goes to 0 and we consider the next insertion as a whole new start.

if it is because of state cleaning due to the watermark, I am not sure.....

Reply:
cannot think of a clean solution for `restarted`.... I guess we don't want to store it into `ResultTable`?

One workaround is to only calculate the changes between two barriers instead of accumulation, and then use `promql` do accumulation on the Grafana side.

The error bound is +/- changes that happened during one checkpoint.